apiserver/internal/handlers/resource/doc.go

```diff
@@ -0,0 +1,7 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package resource contains the HTTP resource endpoint handlers. These are used
+// to transfer the resource blobs between the controller and other entities.
+
+package resource
```

Comment:
nit: Moving the resource handler to internal would have been better as a separate PR.

Reply:
Agreed, I did it here because I wanted to test it, but in retrospect, it should have been done before any of the other changes.